posts/modular-monolith-microservices.md

```diff
@@ -0,0 +1,78 @@
+
+It is essential to recognize that breaking down a monolith and developing microservices can be expensive and time-consuming. Understanding the benefits of transitioning to microservices and why other organizations are doing the same is vital. Some reasons companies should leverage microservices may include the following: 
+
+- Continuous deployment & continuous integration
+- Testing becomes easier with smaller components to test
+- Ability to switch tech stacks over time
+- Teams can develop & test their components without dependency on other teams
+
+Microservices follow the idea of segregation of duty, isolating data to the service that owns it. Each service should be created for a single responsibility, should be independent of others, & needs the ability to handle service-to-service communication in the event a single service fails.
+
+While proven to be incredibly cost-efficient over time, microservices do not come without their challenges. Maintaining & testing data between services can be difficult, as applications that follow this architecture model can become increasingly complex. Fortunately, some services and applications help companies manage, develop, and deploy microservices for both back and front-end needs.
+
+Modular architecture is no longer new and has proven to succeed for enterprises across the globe. Getting started with maintaining them, however, can be tricky for certain companies that don't know where to begin.  Today we will take a look at services that organizations can utilize for microservice deployment, creation, and management.
+
+## Entando
+
+Entando is an open-source Application Composition Platform that allows users to create, curate, and compose applications for business capabilities. Entando is built on top of Kubernetes which allows developers to not only build their applications but run and deploy them as well. There are various components users can create such as microservices, micro front ends, UI/UX widgets, page templates, and so on. Entando seeks to remove redundant, repetitive code and empower developers to reuse code that can be used across applications.
+
+Building a bridge between business and development, Entando uses a 4 step process; create, curate, compose, and consume.  
+
+### Create
+
+Application building with Entando starts with designing and creating independent components. Developers can leverage the [Entando Component Generator](https://developer.entando.com/v7.1/docs/create/component-gen-overview.html) to package and automate project files.
+
+### Curate
+
+Teams can curate by bundling components and sharing them within a central catalog. Several teams can collaborate on individual components or packaged business capabilities (PBCs) while sharing them across organizations and applications. From here they can test components, manage versions and share metadata for future upgrades.
```

Comment:
I think a diagram or link to a diagram would be good here

Reply:
I don't have a diagram for this specific paragraph but just realized my other 2 diagrams had incorrect route so there isn't just text for the whole article thankfully